app/models/piece.rb

```diff
@@ -105,4 +103,26 @@ def move_to!(new_x, new_y)
       self.update(:x_position => new_x, :y_position => new_y)
     end
   end
+
+  def friendly_piece?(new_x, new_y)
+    if !obstacle?(new_x, new_y)
+      return false
+    else
+      obstacle_piece = Piece.find_by(
```

Comment:
@leamerluck This line can be changed to `obstacle_piece = game.pieces.find_by(` so that it only grabs a piece from the current game, not any game.